doc/reference/storage_pure.md

```diff
@@ -0,0 +1,100 @@
+(storage-pure)=
+# PureStorage - `pure`
+
+[PureStorage](https://www.purestorage.com/) is a software-defined storage solution. It offers the consumption of redundant block storage across the network.
+
+LXD offers access to PureStorage storage clusters using either iSCSI or NVMe/TCP.
+In addition, PureStorage offers copy-on-write snapshots, thin provisioning, and other features.
+
+To use PureStorage, ensure the required kernel modules for the selected protocol are installed on your host system.
+For iSCSI, the iSCSI CLI named `iscsiadm` needs to be installed in addition to the required kernel modules.
+
+## Terminology
+
+Each storage pool created in LXD using a PureStorage driver represents a PureStorage *pod*, which is an abstraction that groups multiple volumes under a specific name.
+Another benefit of using PureStorage pods is that they can be linked with multiple PureStorage arrays to provide additional redundancy.
+LXD creates volumes within a pod that is identified by the storage pool name.
+When the first volume needs to be mapped to a specific LXD host, a corresponding PureStorage host is created with the name of the LXD host and a suffix of used protocol.
+The PureStorage host is then connected with the required volumes, to allow attaching and accessing volumes from the LXD host.
+Created PureStorage hosts are automatically removed once there is no volumes connected to it.
+
+## `pure` driver in LXD
+
+The `pure` driver in LXD uses PureStorage volumes for custom storage volumes, instances, and snapshots.
+All created volumes are thin-provisioned block volumes. If required (for example, for containers and custom file system volumes), LXD formats the volume with a desired file system.
+
+LXD expects PureStorage to be pre-configured with a specific service (e.g. iSCSI) on network interfaces whose address is provided during storage pool configuration.
+Furthermore, LXD assumes that it has full control over the PureStorage pods it manages.
+Therefore, you should never maintain any volumes in PureStorage pods that are not owned by LXD because LXD might disconnect or even delete them.
+
+This driver behaves differently than some of the other drivers in that it provides remote storage.
+As a result, and depending on the internal network, storage access might be a bit slower compared to local storage.
+On the other hand, using remote storage has significant advantages in a cluster setup: all cluster members have access to the same storage pools with the exact same contents, without the need to synchronize them.
+
+When creating a new storage pool using the `pure` driver in either `iscsi` or `nvme-tcp` mode, LXD tries to discover one of the array's qualified names on the address `pure.array.address`.
```

Comment:
To avoid confusion if we go for `nvme-fc` in the future. But I agree, having it consistent with PowerFlex would make more sense. Will update everywhere.

Reply:
would there also be potentially an iscsi-fc?

Reply:
There is `scsi-fc` (note missing `i` :))

Reply:
So does pure call nvme over tcp "nvme-tcp"?

Reply:
Yes.

Those are values in PureStorage:
- `nvme-tcp`
- `nvme-fc`
- `scsi-fc`
- `iscsi`

Reply:
@mionaalex shall we go for consistency with powerflex and have "nvme" mode to mean "nvme over tcp" for both powerflex and pure, or use pure specific terminology of "nvme-tcp" and be different from powerflex driver?

Reply:
@tomponline I'd go for consistency, so either we change Pure, or we change PowerFlex as well to be `nvme-tcp` (don't know if that's feasible at this point). If we go for the former, we can still use `nvme-fc` for fiber channel to distinguish between the two.

Reply:
we shouldnt change powerflex as its published and would break deployments, so "nvme" for pure it is.

Reply:
Perfect, thank you both. Will update accordingly.